cfme/infrastructure/host.py

```diff
@@ -452,7 +448,12 @@ class All(CFMENavigateStep):
     prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
 
     def step(self):
-        self.prerequisite_view.navigation.select('Compute', 'Infrastructure', 'Hosts')
+        from cfme.infrastructure.provider.openstack_infra import OpenstackInfraProvider
+        dest = 'Hosts'
+        if hasattr(self.obj, 'provider') and self.obj.provider:
```

Comment:
do you mean to wrap actual navigation into try/except? for example:
`try:
    self.prerequisite_view.navigation.select('Compute', 'Infrastructure', 'Hosts')
except NavigationDestinationError:
    self.prerequisite_view.navigation.select('Compute', 'Infrastructure', 'Nodes')`